docs-developer/custom-importer.md

```diff
@@ -0,0 +1,76 @@
+
+## Useful Docs
+
+ * [Gecko profile format docs](./gecko-profile-format.md)
+ * [processed profile format](./process-profile-format)
+
+## Useful code links
+
+ * [Gecko profile format type definition](../src/types/gecko-profile.js)
+ * [processed profile format type definition](../src/types/profile.js)
+ * [marker payload type definitions](../src/types/markers.js)
+ * [profiler data structure utilities](../src/profile-logic/data-structures.js)
+ * [existing importers](../src/profile-logic/import)
+
+## How to write a profile converter
+
+The rest of this guide will assume you are targeting the processed profile format. From a high level point of view, the profile is broken down into the basic profile, with meta information, and a list of threads. These threads are grouped in the UI by their PID. The UI also dynamically generates a few "tracks" from other data sources, like the memory track or screenshot track. This guide primarily documents working with threads.
+
+A good place to start would probably be to view some of the existing [blank profile generating functions](../src/profile-logic/data-structures.js).
+
+## Some concepts used in the data structures
+
+The profile format uses a lot of indexes into other data structures. For instances, this is how you would extract out a function name from a profile. (This should work from the web console when viewing a profile on profiler.firefox.com.)
+
+```js
+{
+  // Get the first sample out of the first thread:
+  const threadIndex = 0;
+  const sampleIndex = 0;
+
+  const thread = profile.threads[threadIndex];
+
+  // Look up the stack index from the samples table.
+  const stackIndex = thread.samples.stack[sampleIndex];
+  console.log({stackIndex})
+
+  // Look up the frame from the stack table.
+  const frameIndex = thread.stackTable.frame[stackIndex];
+  console.log({frameIndex})
+
+  // Look up the function from the frame table.
+  const funcIndex = thread.frameTable.func[frameIndex];
+  console.log({funcIndex})
+
+  // Look up the string index from the func table.
+  const stringIndex = thread.funcTable.name[funcIndex];
+  console.log({stringIndex})
+
+  console.log(
+    'Function name of the first sample:',
+    thread.stringTable.getString(stringIndex)
+  );
+}
+```
+
+It is probably to read up some on profile format docs for more information on how this structure is used.
+
+## Tips
+
+ * Use `{ name: 'GeckoMain' }` for the main thread.
+ * Ensure that the `pid` value is numeric, and points to the proper threads to nest threads in the timeline.
+ * Processed profiles have their timestamps adjusted so that all processes use the same timeline.
+ * Gecko profiles adjust the marker and sample start times based on when the process was created (based off of `geckoProfile.meta.startTime`).
```

Comment:
I'm not sure this last line is true. Or I don't understand it properly.
I think we use the `startTime` only at processing time, to adjust the difference between processes (this is your 3rd tip and it's good). But then we rely only on the sample time, and don't look at startTime again (see profile-data's `getTimeRangeIncludingAllThreads` that's used to compute `rootRange`).


Also it's not a "tip" but a fact and it's not easy for somebody that reads this to decide what to do out of this sentence.